File: GauchoGrubAzure/GauchoGrub/Models/MenuItemType.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace GauchoGrub.Models
{
    /*
     * MenuItemType - model for representing food item types.
     * Type examples: Regular, Vegetarian, Vegan
     */
    [DataContract]
    public class MenuItemType
    {
        public int Id { get; set; }

        [Required]
        [DataMember]
        [Index(IsUnique = true)] // Extra duplication prevention
        [MaxLength(450)]         // Unique constraint won't work without length limitation
        public String Name { get; set; }
    }
}

File: GauchoGrubAndroid/app/src/main/java/com/g10/gauchogrub/menu/MenuItemType.java
package com.g10.gauchogrub.menu;

import java.util.HashMap;

public class MenuItemType extends NamedEnum {
    public String getShortVersion() {
        HashMap<String, String> map = new HashMap<>();
        map.put("Vegetarian", "(v)");
        map.put("Vegan", "(vgn)");
        map.put("Regular", "");

        return (map.containsKey(this.name)) ? map.get(this.name) : "";
    }
}
